parsel/utils.py

```diff
@@ -55,6 +55,23 @@ def _is_listlike(x):
     """
     return hasattr(x, "__iter__") and not isinstance(x, (six.text_type, bytes))
 
+def extract_regex_first(regex, text):
+    """Extract the first match of unicode strings from the given text/encoding using the following policies:
+    * if the regex contains a named group called "extract" that will be returned
+    * if the regex contains multiple numbered groups, all those will be returned (flattened)
+    * if the regex doesn't contain any group the entire regex matching is returned
+    """
+    if isinstance(regex, six.string_types):
+        regex = re.compile(regex, re.UNICODE)
+
+    m = regex.search(text)
+
+    if m:
+        if 'extract' in regex.groupindex:
+            return replace_entities(m.group('extract'), keep=['lt', 'amp'])
+
+        return [replace_entities(group, keep=['lt', 'amp']) for group in m.groups()]
```

Comment:
No problem. It's not a critical fix or anything. It's the first time I'm using pytest-benchmarks, mostly included it because I wanted to have some proof that what I was optimizing had some effect. I am not 100% certain my changes to the tox/pytest configs are correct, so I'd welcome some review there.

I'll doublecheck with the flatten. I think the function had some odd definitions on what it would return depending on circumstances. _re_replace_entities should be doable too, I'll do a commit when I get the time.